catroid/src/org/catrobat/catroid/utils/DownloadUtil.java

```diff
@@ -124,4 +120,17 @@ protected void onReceiveResult(int resultCode, Bundle resultData) {
 		}
 	}
 
+	public String getProjectNameFromUrl(String url) {
+		int projectNameIndex = url.lastIndexOf(PROJECTNAME_TAG) + PROJECTNAME_TAG.length();
+		String programName = url.substring(projectNameIndex);
+		try {
+			programName = URLDecoder.decode(programName, "UTF-8");
+		} catch (UnsupportedEncodingException e) {
+			Log.e(TAG, "Could not decode program name: " + programName, e);
```

Comment:
If you refactor code, please do not forget about those `e`xception variable shortcuts next time. :ear_of_rice: